### ISSUE TITLE
Missing TrackingOptionID On Invoice LineItem 

### ISSUE THREAD

**Issue by jonny64bit** [XeroAPI/Xero-NetStandard#267]
**SDK you're using (please complete the following information):**
 - Version 3.9.0

**Describe the bug**
When getting invoice with Tracking Category and Tracking Option set on a line item the following is beting returned:

- Name (string)
- Option (string)
- TrackingCategoryID (guid)

However TrackingOptionID is null

**To Reproduce**
1. Call GetInvoicesAsync on the accounting api

**Expected behavior**
I would of expected TrackingOptionID to be present.

**Screenshots**
![image](https://user-images.githubusercontent.com/3177463/100639225-cfa1d800-332c-11eb-8d35-ed89fb85b266.png)

**Comment by SidneyAllen** [comment]
@jenksguo @jamie-hoffmann-xero @JRising-Xero 

I would recommend replicating the call with similar data and reviewing the API logs to confirm the TrackingOptionID is being returned and that the property name and casing matches.  That may be the reason for a null value during deserialization.

**Comment by jamie-hoffmann-xero** [comment]
Hi @jonny64bit, unfortunately I was not able to reproduce the issue.
![image](https://user-images.githubusercontent.com/74579854/100814517-efa0db00-3495-11eb-8b08-0700651205c6.png)

Here's some context about this invoice:
- Using NZ demo company
- Invoice created through the API
- Tracking info was set as per this article: https://central.xero.com/s/article/Track-transactions?userregion=true#Trackprevioustransactions
- Retrieved through the `GetInvoiceAsync` method -- note that this is the singular invoice method

Please let us know how your situation differs from the above, as that may help in tracking down possible causes for the issue.

**Comment by jonny64bit** [comment]
@jamie-hoffmann-xero Thanks for looking into this.

I'm using the UK demo company for this.

What im doing is getting the invoices either for a single client or a batch of clients. Using either of these:

![Get Invoices Screenshot](https://user-images.githubusercontent.com/3177463/100857151-430a2d80-3484-11eb-8ce8-0cca548e9612.PNG)

After this im looping through each of the line items for invoice i can (aka not voided, paid etc.) and then setting the tracking item.

![looping through](https://user-images.githubusercontent.com/3177463/100858892-777ee900-3486-11eb-88a2-628d4733a3b3.PNG)

Then after this im updating invoices singluary using

![update](https://user-images.githubusercontent.com/3177463/100857256-659c4680-3484-11eb-95f7-ea132d5b305b.PNG)

Next time i get the invoice i can see the tracking option has been applied but the tracking option id is missing

![full debug view of error](https://user-images.githubusercontent.com/3177463/100859007-9e3d1f80-3486-11eb-858b-81f2fefcb75d.png)

Doesn't make a difference if its on a invoice the demo company has created or one i have created.

Both my examples use the `GetInvoicesAsync` so maybe this is it?

Is there away for me to get more logging out of the client so i can inspect the raw response?

Mayeb worth mentioning this is all within a .Net 5 IHostedService running in docker.

**Comment by jamie-hoffmann-xero** [comment]
Hi @jonny64bit, thanks for the quick response. I'll look into this a bit today.

I suspect using the `GetInvoicesAsync` method might be related -- the endpoints that return multiple items often don't return the full model, though your case is unusually specific.

In the meantime, you can use the `GetInvoicesAsyncWithHttpInfo` method in order to inspect the raw response. This method will return an `ApiResponse<Invoices>` containing information about the response as well as the requested data:
![image](https://user-images.githubusercontent.com/74579854/100956747-44635500-356d-11eb-8837-a29e4d3b21a5.png)

**Comment by jamie-hoffmann-xero** [comment]
Just an update on this:
After investigation, I've found that using `GetInvoicesAsync` filtered by contact ID returns tracking info, but instead of a `TrackingOptionID`, we get an `Options` array in the response instead. For me, the array is empty.

Unfortunately, this is outside the scope of the SDK, but I've contacted the API team and will report back with findings.
For now, the `GetInvoiceAsync` method will return `TrackingOptionID` as expected.

Thanks for reporting this one, @jonny64bit
